resources/bundles/org.eclipse.core.filesystem/src/org/eclipse/core/internal/filesystem/local/LocalFile.java

```diff
@@ -246,36 +270,44 @@ private boolean internalDelete(File target, String pathToDelete, MultiStatus sta
 				childBuffer.append(File.separatorChar);
 				childBuffer.append(element);
 				String childName = childBuffer.toString();
-				// Try best effort on all children so put logical OR at end.
-				failedRecursive = !internalDelete(new java.io.File(childName), childName, status, monitor) || failedRecursive;
+
+				deleteChildrenStatus.add(internalDelete(new java.io.File(childName), childName, monitor));
+
 				monitor.worked(1);
 			}
+
+			// Abort if one of the children couldn't be deleted.
+			if (!deleteChildrenStatus.isOK())
+				return deleteChildrenStatus;
+
```

Comment:
Lets add some braces here too.
```suggestion
			if (!deleteChildrenStatus.isOK()) {
				return deleteChildrenStatus;
			}
```